diff --git a/configure b/configure
index f43a669fb..14dbd0d0d 100755
--- a/configure
+++ b/configure
@@ -496,7 +496,7 @@ $arch_compilerc   = $arch_echo;
 $arch_rcflags     = "No resource compiler required on this platform.";
 
 $arch_cc          = "cc";
-$arch_ccpp        = "CC";
+$arch_ccpp        = "CC -std=c++11";
 $arch_nvcc        = "/usr/local/cuda-10.2/bin/nvcc";
 $arch_nvccflags   = "-lineinfo --ptxas-options=-v " . 
                     "-gencode arch=compute_30,code=compute_30 " .
@@ -511,7 +511,7 @@ $arch_nvccflags   = "-lineinfo --ptxas-options=-v " .
                     "--ftz=true ";
 #                    "-gencode arch=compute_75,code=sm_75 " .
 $arch_gcc         = "gcc";
-$arch_gccpp       = "g++";
+$arch_gccpp       = "g++ -std=c++11";
 $arch_lex         = "lex";
 $arch_yacc        = "yacc";
 $arch_yflags      = "-d";
@@ -2146,7 +2146,7 @@ if ($config_arch eq "ANDROIDARMV7A") {
   $androidsysroot = "--sysroot=/home/kvandivo/Proj/android-ndk-r8b/platforms/android-14/arch-arm";
   $androidabi = "arm-linux-androideabi-";
   $arch_cc       = $androidbins . $androidabi . "gcc";
-  $arch_ccpp     = $androidbins . $androidabi . "g++";
+  $arch_ccpp     = $androidbins . $androidabi . "g++ -std=c++11";
   $arch_copts       = "-Wall -Wno-unknown-pragmas -Wno-psabi -MMD -MP -fpic -ffunction-sections -funwind-tables -fstack-protector -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE__ -march=armv7-a -msoft-float -mfpu=vfp -fno-exceptions -mthumb -Os -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 -Wa,--noexecstac";
   $arch_opt_flag    = "-Wall -Wno-unknown-pragmas -Wno-psabi -MMD -MP -fpic -ffunction-sections -funwind-tables -fstack-protector -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE__ -march=armv7-a -msoft-float -mfpu=vfp -fno-exceptions -mthumb -Os -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 -Wa,--noexecstac";
   $arch_copts    = $androidincs;
@@ -2199,7 +2199,7 @@ if ($config_arch eq "BLUEWATERS" ||
   if ($config_icc) {
     # for compiling with Intel C/C++:
     $arch_cc          = "cc";
-    $arch_ccpp        = "CC";
+    $arch_ccpp        = "CC -std=c++11";
     $arch_copts       = "-Wall -Wno-unknown-pragmas -m64 -msse2 -O3";
     $arch_opt_flag    = "-Wall -Wno-unknown-pragmas -m64 -msse2 -O3";
 
@@ -2216,9 +2216,9 @@ if ($config_arch eq "BLUEWATERS" ||
   } else {
     # Assume GCC 
     $arch_cc          = "cc";
-    $arch_ccpp        = "CC";
+    $arch_ccpp        = "CC -std=c++11";
     $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O6 -ffast-math";
-    $arch_opt_flag    = "-m64 -fno-for-scope -Wno-deprecated -Wall -Wno-unknown-pragmas -O6 -ffast-math";
+    $arch_opt_flag    = "-m64 -Wno-deprecated -Wall -Wno-unknown-pragmas -O6 -ffast-math";
   }
 
   if ($config_mpi) {
@@ -2320,14 +2320,14 @@ if ($config_arch eq "FREEBSD") {
     $arch_yacc     = "bison -b y -d ";  # If standard yacc doesn't work
 
     $arch_cc          = "gcc";
-    $arch_ccpp        = "g++";
+    $arch_ccpp        = "g++ -std=c++11";
     $arch_depend_flag = "-MM";
     $arch_shld        = "g++ -shared";
     $arch_shlibname   = "so";
     $arch_shcppopts   = "-fPIC";
     $arch_shldopts    = "-L/usr/local/lib";
 
-    $arch_opt_flag    = "-m32 -fno-for-scope -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
+    $arch_opt_flag    = "-m32 -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
     $arch_copts       = "-m32 -Wall -Wno-unknown-pragmas -O3";
 
     if ($config_static) {
@@ -2376,7 +2376,7 @@ if ($config_arch eq "FREEBSDAMD64") {
     $arch_yacc     = "bison -b y -d ";  # If standard yacc doesn't work
 
     $arch_cc          = "gcc";
-    $arch_ccpp        = "g++";
+    $arch_ccpp        = "g++ -std=c++11";
     $arch_depend_flag = "-MM";
     $arch_shld        = "ld -shared";
     $arch_shlibname   = "so";
@@ -2455,13 +2455,13 @@ if ($config_arch eq "LINUX") {
     } else {
       # compling with GCC
       $arch_cc          = "gcc";
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_depend_flag = "-MM";
       $arch_shld        = "g++ -shared";
       $arch_shlibname   = "so";
       $arch_shcppopts   = "-fPIC";
       $arch_shldopts    = "";
-      $arch_opt_flag    = "-m32 -fno-for-scope -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
+      $arch_opt_flag    = "-m32 -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
       $arch_copts       = "-m32 -Wall -Wno-unknown-pragmas -O3";
 
       if ($config_static) {
@@ -2522,7 +2522,7 @@ if ($config_arch eq "LINUXAMD64") {
     if ($config_icc) {
       # for compiling with Intel C/C++:
       $arch_cc          = "icc";
-      $arch_ccpp        = "icpc";
+      $arch_ccpp        = "icpc -std=c++11";
       if ($config_avx512) {
         $arch_opt_flag    = "-Wall -Wno-unknown-pragmas -m64 -xMIC-AVX512 -O3";
         $arch_copts       = "-Wall -Wno-unknown-pragmas -m64 -xMIC-AVX512 -O3";
@@ -2564,15 +2564,15 @@ if ($config_arch eq "LINUXAMD64") {
       }
     } elsif ($config_clang) {
       $arch_cc          = "clang";
-      $arch_ccpp        = "clang++";
+      $arch_ccpp        = "clang++ -std=c++11";
       $arch_opt_flag    = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math ";
       $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math ";
       $arch_depend_flag = "-MM";
     } else {
       # for compiling with g++:
       $arch_cc		= "gcc";
-      $arch_ccpp  	= "g++";
-      $arch_opt_flag	= "-m64 -fno-for-scope -Wno-deprecated -Wall -Wno-unknown-pragmas -O6 -ffast-math";
+      $arch_ccpp  	= "g++ -std=c++11";
+      $arch_opt_flag	= "-m64 -Wno-deprecated -Wall -Wno-unknown-pragmas -O6 -ffast-math";
       $arch_depend_flag   = "-MM";
       $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O6 -ffast-math";
 
@@ -2629,7 +2629,7 @@ if ($config_arch eq "LINUXARM64") {
                     "--ftz=true ";
   $cuda_library     = "-L/usr/local/cuda/lib64";
   $arch_cc       = "cc";
-  $arch_ccpp     = "c++";
+  $arch_ccpp     = "c++ -std=c++11";
   $arch_copts       = "-Wall -Wno-unknown-pragmas -ffast-math -fomit-frame-pointer -fno-strict-aliasing -O3";
   $arch_opt_flag    = "-Wall -Wno-unknown-pragmas -ffast-math -fomit-frame-pointer -fno-strict-aliasing -O3";
   $arch_lex      = "flex";
@@ -2656,7 +2656,7 @@ if ($config_arch eq "LINUXCARMA") {
                     "--ftz=true ";
   $cuda_library     = "-L/usr/local/cuda/lib";
   $arch_cc       = "cc";
-  $arch_ccpp     = "c++";
+  $arch_ccpp     = "c++ -std=c++11";
   $arch_copts       = "-Wall -Wno-unknown-pragmas -march=armv7-a -mcpu=cortex-a9 -mtune=cortex-a9 -mfpu=vfpv3-d16 -ffast-math -fomit-frame-pointer -fno-strict-aliasing -O3";
   $arch_opt_flag    = "-Wall -Wno-unknown-pragmas -march=armv7-a -mcpu=cortex-a9 -mtune=cortex-a9 -mfpu=vfpv3-d16 -ffast-math -fomit-frame-pointer -fno-strict-aliasing -O3";
   $arch_lex      = "flex";
@@ -2674,7 +2674,7 @@ if ($config_arch eq "LINUXPPC64") {
     }
 
     $arch_cc          = "gcc";
-    $arch_ccpp        = "g++";
+    $arch_ccpp        = "g++ -std=c++11";
     $arch_depend_flag = "-MM";
     $arch_opt_flag    = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math";
     $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math";
@@ -2714,7 +2714,7 @@ if ($config_arch eq "MACOSX") {
     $python_libs        = "-framework Python -lpthread";
 
     $arch_cc          = "cc";
-    $arch_ccpp        = "c++";
+    $arch_ccpp        = "c++ -std=c++11";
     $arch_depend_flag = "-MM";
     $arch_opt_flag    = "-fPIC -Os -ffast-math";
     $arch_copts       = "-fPIC -Os -ffast-math";
@@ -2799,7 +2799,7 @@ if ($config_arch eq "MACOSXARM64") {
     $python_libs        = "-framework Python -lpthread";
 
     $arch_cc          = "cc";
-    $arch_ccpp        = "c++";
+    $arch_ccpp        = "c++ -std=c++11";
     $arch_depend_flag = "-MM";
     $arch_opt_flag    = "-fPIC -Os -ffast-math";
     $arch_copts       = "-fPIC -Os -ffast-math";
@@ -2905,13 +2905,13 @@ if ($config_arch eq "MACOSXX86" || $config_arch eq "MACOSXX86_64") {
 
     if ($config_lp64) {
       $arch_cc          = "cc";
-      $arch_ccpp        = "c++";
+      $arch_ccpp        = "c++ -std=c++11";
       $arch_depend_flag = "-MM";
       $arch_opt_flag    = "-m64 -fPIC -Os -ffast-math";
       $arch_copts       = "-m64 -fPIC -Os -ffast-math";
     } else {
       $arch_cc          = "cc";
-      $arch_ccpp        = "c++";
+      $arch_ccpp        = "c++ -std=c++11";
       $arch_depend_flag = "-MM";
       $arch_opt_flag    = "-m32 -fPIC -Os -ffast-math";
       $arch_copts       = "-m32 -fPIC -Os -ffast-math";
@@ -3043,20 +3043,20 @@ if ($config_arch eq "SUMMIT") {
 
     if ($config_xlc) {
       $arch_cc          = "xlc";
-      $arch_ccpp        = "xlC";
+      $arch_ccpp        = "xlC -qlanglvl=extended0x";
       $arch_copts       = "-O4 -qsuppress=1500-036 -q64 -qarch=pwr9 -qtune=pwr9 -qmaxmem=-1 -qnoipa -qignerrno -qfloat=rsqrt -qnostrict";
       $arch_opt_flag    = "-O4 -qsuppress=1500-036 -q64 -qarch=pwr9 -qtune=pwr9 -qmaxmem=-1 -qnoipa -qignerrno -qfloat=rsqrt -qnostrict";
 #      $arch_cppopts     = "-D_H_ACCESS=1";
       $arch_depend_flag = "-MM";
     } elsif ($config_gcc) {
       $arch_cc          = "gcc";
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_opt_flag    = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power8";
       $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power8";
       $arch_depend_flag = "-MM";
     } elsif ($config_clang) {
       $arch_cc          = "clang";
-      $arch_ccpp        = "clang++";
+      $arch_ccpp        = "clang++ -std=c++11";
       $arch_opt_flag    = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power9";
       $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power9";
       $arch_depend_flag = "-MM";
@@ -3116,14 +3116,14 @@ if ($config_arch eq "OPENPOWER" || $config_arch eq "SUMMITDEV") {
 
     if ($config_xlc) {
       $arch_cc          = "xlc";
-      $arch_ccpp        = "xlC";
+      $arch_ccpp        = "xlC -qlanglvl=extended0x";
       $arch_copts       = "-O4 -q64 -qarch=pwr8 -qtune=pwr8 -qmaxmem=-1 -qnoipa -qignerrno -qfloat=rsqrt -qnostrict";
       $arch_opt_flag    = "-O4 -q64 -qarch=pwr8 -qtune=pwr8 -qmaxmem=-1 -qnoipa -qignerrno -qfloat=rsqrt -qnostrict";
 #      $arch_cppopts     = "-D_H_ACCESS=1";
       $arch_depend_flag = "-MM";
     } else {
       $arch_cc          = "gcc";
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_opt_flag    = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power8";
       $arch_copts       = "-m64 -Wall -Wno-unknown-pragmas -O3 -ffast-math -mtune=power8";
       $arch_depend_flag = "-MM";
@@ -3176,7 +3176,7 @@ if ($config_arch eq "SOLARIS2" || $config_arch eq "SOLARIS2_64") {
       if ($config_lp64) {
         print "GCC doesn't support LP64 ABI on this platform currently.\n";
       }
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_cc	        = "gcc";
       $arch_opt_flag    = "-O";
       $arch_copts       = "-O";
@@ -3186,7 +3186,7 @@ if ($config_arch eq "SOLARIS2" || $config_arch eq "SOLARIS2_64") {
       # Using Sun's compilers.
       if ($config_lp64) {
         print "Configured for a 64-bit build.\n";
-        $arch_ccpp        = "CC -D__EXTENSIONS__";
+        $arch_ccpp        = "CC -D__EXTENSIONS__ -std=c++11";
         $arch_cc          = "cc -D__EXTENSIONS__";
         if ($config_pthreads) {
           $arch_opt_flag    = "-mt -fast -xO3 -native -xarch=v9a";
@@ -3199,7 +3199,7 @@ if ($config_arch eq "SOLARIS2" || $config_arch eq "SOLARIS2_64") {
         $arch_depend_flag = "-xM";
       } else {
         print "Configured for a 32-bit build.\n";
-        $arch_ccpp        = "CC -D__EXTENSIONS__";
+        $arch_ccpp        = "CC -D__EXTENSIONS__ -std=c++11";
         $arch_cc          = "cc -D__EXTENSIONS__";
         if ($config_pthreads) {
           $arch_opt_flag    = "-mt -fast -xO3 -native -xarch=v8plusa";
@@ -3249,7 +3249,7 @@ if ($config_arch eq "SOLARISX86" || $config_arch eq "SOLARISX86_64") {
       if ($config_lp64) {
         die "No LP64 gcc for this platform currently.\n";
       }
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_cc	        = "gcc";
 # XXX gcc/g++ 2.95 and newer need -fpermissive since the X11 headers 
 # on Solaris (and other systems) are broken and still assume that missing
@@ -3263,7 +3263,7 @@ if ($config_arch eq "SOLARISX86" || $config_arch eq "SOLARISX86_64") {
       # Using Sun's compilers
       if ($config_lp64) {
         print "Configured for a 64-bit build.\n";
-        $arch_ccpp        = "CC -D__EXTENSIONS__";
+        $arch_ccpp        = "CC -D__EXTENSIONS__ -std=c++11";
         $arch_cc          = "cc -D__EXTENSIONS__";
         if ($config_pthreads) {
           $arch_opt_flag    = "-mt -fast -xO5 -xarch=native64";
@@ -3284,7 +3284,7 @@ if ($config_arch eq "SOLARISX86" || $config_arch eq "SOLARISX86_64") {
         }
       } else {
         print "Configured for a 32-bit build.\n";
-        $arch_ccpp        = "CC -D__EXTENSIONS__";
+        $arch_ccpp        = "CC -D__EXTENSIONS__ -std=c++11";
         $arch_cc          = "cc -D__EXTENSIONS__";
         if ($config_pthreads) {
           $arch_copts       = "-mt -fast -xO3 -native";
@@ -3331,7 +3331,7 @@ if ($config_arch eq "SOLARISX86" || $config_arch eq "SOLARISX86_64") {
 
 if ($config_arch eq "WIN32") {
     if ($config_msvc) {
-      $arch_ccpp        = "cl.exe";
+      $arch_ccpp        = "cl.exe /std:c11";
       $arch_cc	        = "cl.exe";
       $arch_opt_flag    = "/O2 /G5 /nologo /MT /DWIN32 -Dstrcasecmp=strupcmp";
       $arch_debug_flag  = "-g";
@@ -3340,7 +3340,7 @@ if ($config_arch eq "WIN32") {
       $arch_cppopts     = "/TP";
       $arch_coptout     = "/Fo";
     } else {
-      $arch_ccpp        = "g++";
+      $arch_ccpp        = "g++ -std=c++11";
       $arch_cc	        = "gcc";
       $arch_opt_flag    = "-O";
       $arch_debug_flag  = "-g";
@@ -3357,7 +3357,7 @@ if ($config_arch eq "WIN32") {
 
 
 if ($config_arch eq "WIN64") {
-    $arch_ccpp	      = "g++";
+    $arch_ccpp	      = "g++ -std=c++11";
     $arch_cc	      = "gcc";
     $arch_opt_flag    = "-O";
     $arch_debug_flag  = "-g";
