diff --git a/configure b/configure
index 70fba19bf..997b06046 100755
--- a/configure
+++ b/configure
@@ -202,7 +202,7 @@ $config_actc            = 0;  # ACTC
 $config_avx512          = 0;  # AVX512
 $config_cpudispatch     = 0;  # CPUDISPATCH
 $config_cuda            = 0;  # CUDA
-$config_cxx11           = 0;  # Require C++11 or greater
+$config_cxx11           = 1;  # Require C++11 or greater
 $config_opencl          = 0;  # OpenCL
 $config_openhmd         = 0;  # OpenHMD driver for Oculus Rift
 $config_imd             = 0;  # interactive MD sockets code etc.
@@ -2411,7 +2411,7 @@ if ($config_arch eq "LINUX") {
       $arch_shlibname   = "so";
       $arch_shcppopts   = "-fPIC";
       $arch_shldopts    = "";
-      $arch_opt_flag    = "-m32 -fno-for-scope -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
+      $arch_opt_flag    = "-m32 -Wno-deprecated -Wall -Wno-unknown-pragmas -O3";
       $arch_copts       = "-m32 -Wall -Wno-unknown-pragmas -O3";
 
       if ($config_static) {
@@ -2536,7 +2536,10 @@ if ($config_arch eq "LINUXAMD64") {
     }
 
     if ($config_cxx11) {
-      $arch_opt_flag .= " -std=c++11"
+      $arch_opt_flag .= " -std=c++11";
+      if ($config_cuda) {
+        $arch_nvccflags .= " -std=c++11";
+      }
     }
 
     if ($config_cuda) {
